lib/oneview-sdk/resource_helper.rb

```diff
@@ -12,19 +12,33 @@
 module OneviewSDK
   # Contains helper methods to include certain functionalities on resources
   module ResourceHelper
-    # Performs a specific patch operation for the given resource.
-    # If the resource supports the particular operation, the operation is performed
-    # and a response is returned to the caller with the results.
-    # @param [String] operation The operation to be performed
-    # @param [String] path The path of operation
-    # @param [String] value The value
-    # @note This attribute is subject to incompatible changes in future release versions, including redefinition or removal.
-    def patch(operation, path, value = nil, header_options = {})
-      ensure_client && ensure_uri
-      options = { 'body' => [op: operation, path: path, value: value] }
-      options = options.merge(header_options)
-      response = @client.rest_patch(@data['uri'], options, @api_version)
-      @client.response_handler(response)
+    # Contains helper method to call patch endpoint of resource
+    module PatchOperation
+      # Performs a specific patch operation for the given resource.
+      # If the resource supports the particular operation, the operation is performed
+      # and a response is returned to the caller with the results.
+      # @param [String] operation The operation to be performed
+      # @param [String] path The path of operation
+      # @param [String] value The value
+      # @note This attribute is subject to incompatible changes in future release versions, including redefinition or removal.
+      def patch(operation, path, value = nil, header_options = {})
+        ensure_client && ensure_uri
+        options = { 'body' => [op: operation, path: path, value: value] }
+        options = options.merge(header_options)
+        response = @client.rest_patch(@data['uri'], options, @api_version)
+        @client.response_handler(response)
+      end
+    end
+
+    # Contains helper method to call configuration endpoint of resource
+    module ConfigurationOperation
+      # Reapplies the configuration
+      def configuration
```

Comment:
Perhaps `reapply_configuration` or something similar could be a better name?
I understand that `configuration` is the name of the endpoint, but it is hard to discern what this method does by its name.

Reply:
ok, it makes sense. =) 
The name `reapply_configuration` is used in oneview-chef resources too.
I gonna change that.

Reply:
Or we can keep the `configuration` method and adding the `reapply_configuration` method, the `configuration` can just call the `reapply_configuration` and we would mark the `configuration` as a deprecated. I think that is a good option.

Reply:
@ricardogpsf , to avoid backward compatibility, which is a concern we have with the SDK, it may be the best option. But I think that applying a deprecate in a method to just changing the name may be unnecessary, so I prefer to keep the same name ```configuration```.

Reply:
I guess we don't need to break backward compatibility just because of this

Reply:
ok, maybe we can keep the code how is it in this PR, right?

Reply:
I'm fine with it, the name is not `ideal` but the ship is long sailed and by now I definitely agree it isn't such a big thing to be worth breaking compatibility.
I'd leave it as is. :octocat: